### ISSUE TITLE
Simplify cache smoke test

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#19325]
```
$ ./mvnw -f spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-cache -Pehcache verify

…

[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ spring-boot-smoke-test-cache ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running smoketest.cache.SampleCacheApplicationTests
20:20:55.587 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating CacheAwareContextLoaderDelegate from class [org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate]
20:20:55.600 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating BootstrapContext using constructor [public org.springframework.test.context.support.DefaultBootstrapContext(java.lang.Class,org.springframework.test.context.CacheAwareContextLoaderDelegate)]
20:20:55.625 [main] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating TestContextBootstrapper for test class [smoketest.cache.SampleCacheApplicationTests] from class [org.springframework.boot.test.context.SpringBootTestContextBootstrapper]
20:20:55.638 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Neither @ContextConfiguration nor @ContextHierarchy found for test class [smoketest.cache.SampleCacheApplicationTests], using SpringBootContextLoader
20:20:55.641 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [smoketest.cache.SampleCacheApplicationTests]: class path resource [smoketest/cache/SampleCacheApplicationTests-context.xml] does not exist
20:20:55.642 [main] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [smoketest.cache.SampleCacheApplicationTests]: class path resource [smoketest/cache/SampleCacheApplicationTestsContext.groovy] does not exist
20:20:55.642 [main] INFO org.springframework.test.context.support.AbstractContextLoader - Could not detect default resource locations for test class [smoketest.cache.SampleCacheApplicationTests]: no resource found for suffixes {-context.xml, Context.groovy}.
20:20:55.642 [main] INFO org.springframework.test.context.support.AnnotationConfigContextLoaderUtils - Could not detect default configuration classes for test class [smoketest.cache.SampleCacheApplicationTests]: SampleCacheApplicationTests does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
20:20:55.674 [main] DEBUG org.springframework.test.context.support.ActiveProfilesUtils - Could not find an 'annotation declaring class' for annotation type [org.springframework.test.context.ActiveProfiles] and class [smoketest.cache.SampleCacheApplicationTests]
20:20:55.747 [main] DEBUG org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider - Identified candidate component class: file [/Users/awilkinson/dev/spring-projects/spring-boot/master/spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-cache/target/classes/smoketest/cache/SampleCacheApplication.class]
20:20:55.751 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Found @SpringBootConfiguration smoketest.cache.SampleCacheApplication for test class smoketest.cache.SampleCacheApplicationTests
20:20:55.832 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - @TestExecutionListeners is not present for class [smoketest.cache.SampleCacheApplicationTests]: using defaults.
20:20:55.832 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener, org.springframework.test.context.event.EventPublishingTestExecutionListener]
20:20:55.842 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Skipping candidate TestExecutionListener [org.springframework.test.context.transaction.TransactionalTestExecutionListener] due to a missing dependency. Specify custom listener classes or make the default listener classes and their required dependencies available. Offending class: [org/springframework/transaction/interceptor/TransactionAttributeSource]
20:20:55.843 [main] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Skipping candidate TestExecutionListener [org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener] due to a missing dependency. Specify custom listener classes or make the default listener classes and their required dependencies available. Offending class: [org/springframework/transaction/interceptor/TransactionAttribute]
20:20:55.843 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@121314f7, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@130c12b7, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@5e600dd5, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@576d5deb, org.springframework.test.context.support.DirtiesContextTestExecutionListener@5d534f5d, org.springframework.test.context.event.EventPublishingTestExecutionListener@2e3967ea, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@70e9c95d, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@4d5650ae, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@a38c7fe, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@6fdbe764, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@51c668e3]
20:20:55.846 [main] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - Before test class: context [DefaultTestContext@38cee291 testClass = SampleCacheApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@4b45a2f5 testClass = SampleCacheApplicationTests, locations = '{}', classes = '{class smoketest.cache.SampleCacheApplication}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}', contextCustomizers = set[org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@1ddf84b8, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@5e82df6a, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@3c407114, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@1ab3a8c8], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> true]], class annotated with @DirtiesContext [false] with mode [null].
20:20:55.867 [main] DEBUG org.springframework.test.context.support.TestPropertySourceUtils - Adding inlined properties to environment: {spring.jmx.enabled=false, org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true, server.port=-1}

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::  (v2.3.0.BUILD-SNAPSHOT)

2019-12-06 20:20:56.061  INFO 71654 --- [           main] s.cache.SampleCacheApplicationTests      : Starting SampleCacheApplicationTests on Andys-MacBook-Pro.local with PID 71654 (started by awilkinson in /Users/awilkinson/dev/spring-projects/spring-boot/master/spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-cache)
2019-12-06 20:20:56.062  INFO 71654 --- [           main] s.cache.SampleCacheApplicationTests      : No active profile set, falling back to default profiles: default
2019-12-06 20:20:57.497  INFO 71654 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2019-12-06 20:20:57.841  INFO 71654 --- [           main] o.s.s.c.ThreadPoolTaskScheduler          : Initializing ExecutorService 'taskScheduler'
2019-12-06 20:20:57.856  INFO 71654 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 13 endpoint(s) beneath base path '/actuator'
2019-12-06 20:20:57.904  INFO 71654 --- [           main] s.cache.SampleCacheApplicationTests      : Started SampleCacheApplicationTests in 2.029 seconds (JVM running for 2.749)
2019-12-06 20:20:57.907  INFO 71654 --- [           main] smoketest.cache.CacheManagerCheck        : 

=========================================================
Using cache manager: org.springframework.cache.jcache.JCacheCacheManager
=========================================================


[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.531 s <<< FAILURE! - in smoketest.cache.SampleCacheApplicationTests
[ERROR] validateCache  Time elapsed: 0.155 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting actual not to be null
	at smoketest.cache.SampleCacheApplicationTests.validateCache(SampleCacheApplicationTests.java:40)

2019-12-06 20:20:58.118  INFO 71654 --- [extShutdownHook] o.s.s.c.ThreadPoolTaskScheduler          : Shutting down ExecutorService 'taskScheduler'
2019-12-06 20:20:58.119  INFO 71654 --- [extShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   SampleCacheApplicationTests.validateCache:40 
Expecting actual not to be null
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0

**Comment by wilkinsona** [comment]
I've marked this as a task for now as I'm not sure if:

- I'm missing something about using the smoke test with EhCache
- The smoke test is faulty
- There's a problem with our EhCache integration

The failure occurs with 2.3. I haven't checked earlier branches. It may occur there too.

**Comment by eddumelendez** [comment]
@wilkinsona the reason is because `ehcache3.xml` file is not set. currently, that property is commented as you can see 

https://github.com/spring-projects/spring-boot/blob/d9466f5659435eb8ff73cdb347a86f0ede403cce/spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-cache/src/main/resources/application.properties#L12

**Comment by snicoll** [comment]
Thanks @eddumelendez. Those various profiles are meant to bring the necessary libraries on the classpath. The `README` provides additional instructions as what a user should do if they want to run the test with that library. The goal of the profiles is to let you try different cache provider without having to patch the pom file.

Having said that, now that we've moved those samples to smoke tests, I think all those profiles (and the readme) should go away as they were added for the only purpose of letting users try this out. This doesn't seem to be a valid objective anymore.
